File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/Models/Account.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankingPaymentsApp_API.Models
{
    public class Account
    {
        [Key]
        public int AccountId {  get; set; }
        [Required(ErrorMessage = "Account Number is Required!")]
        [RegularExpression(@"^BPA\d{8}[A-Z0-9]{6}$", ErrorMessage ="Account Number is Not Valid")]
        public string AccountNumber {  get; set; }
        [ForeignKey("ClientUser")]
        public int? ClientId {  get; set; }
        public virtual ClientUser? ClientUser { get; set; }
        [Required(ErrorMessage = "Balance in Required!")]
        [DataType(DataType.Currency)]
        public double Balance { get; set; } = 0;
        [Required(ErrorMessage = "Account Type is Required!")]
        [ForeignKey("AccountType")]
        public int AccountTypeId { get; set; }
        public virtual AccountType? AccountType { get; set; }
        [Required(ErrorMessage = "Account Status is Required!")]
        [ForeignKey("AccountStatus")]
        public int AccountStatusId { get; set; }
        public virtual AccountStatus? AccountStatus { get; set; } 
        [Required(ErrorMessage = "Account Creation Date is Required!")]
        [DataType(DataType.Date)]
        public DateTime CreatedAt {  get; set; } = DateTime.Now.Date;
        public virtual List<int>? TransactionIds { get; set; } = new List<int>();
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/Models/Account.ts
import { AccountStatus } from "./AccountStatus";
import { AccountType } from "./AccountType";
import { Bank } from "./Bank";
import { ClientUser } from "./ClientUser";

export interface Account {
  accountId: number;
  accountNumber: string;
  clientId?: number;
  clientUser?: ClientUser;
  balance: number;
  bankId:number;
  bank:Bank;
  accountTypeId: number;
  accountType?: AccountType;
  accountStatusId: number;
  accountStatus?: AccountStatus;
  createdAt: string; 
  transactionIds?: Array<number>;
}
